feat(tui): fork slash command for keyboard-friendly session forking (resolves #5599)

ariane-emory opened the pull request:
This PR adds a `/fork` slash command which pops a "Fork from message" modal dialog window resembling the existing timeline dialog window. Upon selection of a message in this window, the session is immediately forked upon that message, as if the user had clicked on the message and selected the fork option.

This command may also be bound to a key using the key command name `session_fork` if the user chooses to bind it in their `opencode.json` file (but is not bound to any key by default).

<img width="682" height="438" alt="Screenshot 2025-12-16 at 12 25 02 AM" src="https://github.com/user-attachments/assets/e13aa742-d2e9-41c8-a130-6e8d0b3385d7" />

Note: I do think it is slightly awkward that this timeline includes `ignored`/`synthatic` messages from plugins in the list (the DCP messages shown in the screenshot), but filtering them out would make it inconsistent with the behaviour of the existing session timeline, and I do prefer consistency. Addressing this by making both of these timeline dialogs exclude these plugin methods is probably outside the scope of this particular PR, and so I'll leave that as something to consider for the future.

Resolves #5599.

ariane-emory committed:
Add /fork slash command for keyboard-friendly session forking

- Create DialogForkTimeline component that shows timeline and forks on selection
- Add /fork command to autocomplete that opens fork timeline dialog
- Register session.fork command with keybind support (defaults to none)
- Add session_fork keybind config option
- Show toast notification when fork is created

Closes #5599

ariane-emory committed:
Add /fork command for keyboard-accessible session forking

- Add /fork to autocomplete options in prompt
- Register session.fork command to open fork timeline dialog
- Add session_fork keybinding configuration
- Implement DialogForkTimeline component for message selection

Addresses #5599 by enabling keyboard-only users to fork sessions
without needing to use mouse interactions.

ariane-emory committed:
Fix /fork command to directly fork without intermediate modal

- Replace DialogTimeline with DialogForkTimeline in session.fork command
- DialogForkTimeline directly calls sdk.client.session.fork() on message selection
- Fixes issue #5599: /fork now immediately forks after message selection
- Eliminates extra 'fork/copy/revert' modal step that was breaking UX

ariane-emory committed:
Rename dialog-fork-timeline to dialog-fork-from-timeline

- Rename file from dialog-fork-timeline.tsx to dialog-fork-from-timeline.tsx
- Update component name from DialogForkTimeline to DialogForkFromTimeline
- Update imports and usage in session index
- Improves naming clarity to better describe component purpose

ariane-emory committed:
tweak: adjust some text

ariane-emory commented:
Work is proceeding, this is coming along well so far. After I get some caffeine and some breakfast in me, I'm going to spend a couple more hours testing it to make sure everything is airtight and tidy before it comes out of draft.

ariane-emory commented:
LGTM now, let me know if you guys have any thoughts or concerns.

ariane-emory committed:
Merge branch 'dev' into feat/fork-slash-command

ariane-emory commented:
@rekram1-node Any concerns here?

rekram1-node commented:
/review

rekram1-node commented:
I think we may as well make the timeline not show synthetic or ignored ones and then do that here too like why not right?

ariane-emory commented:
@rekram1-node Great, will do. Just to filter them from `/fork`'s timeline, or from the standard `session_timeline` as well? LMK and I'll take care of it presently. :)

ariane-emory committed:
tidy: indentation

ariane-emory committed:
Apply suggestion from @github-actions[bot]

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

rekram1-node commented:
prolly both?

ariane-emory committed:
Apply suggestion from @github-actions[bot]

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ariane-emory committed:
Filter out ignored messages from timeline dialogs

- Add && !x.ignored filter to DialogTimeline to skip ignored parts
- Add && !x.ignored filter to DialogForkFromTimeline to skip ignored parts
- Fix malformed fork command object in session/index.tsx (missing value, duplicate onSelect)
- Add missing session_rename keybind to config.ts

Both /timeline and /fork now filter out synthetic AND ignored messages as suggested by rekram1-node

ariane-emory committed:
tidy: line spacing

ariane-emory committed:
tidy: line spacing

ariane-emory commented:
@rekram1-node Done, `ignored` / `synthetic` messages are now filtered from both modals.

rekram1-node commented:
/review
